include/dynd/types/tuple_type.hpp

```diff
@@ -145,13 +147,13 @@ namespace ndt {
     }
 
     /** Makes a tuple type with the specified types */
-    static type make(const nd::array &field_types, bool variadic = false)
+    static type make(std::vector<type> field_types, bool variadic = false)
     {
       return type(new tuple_type(field_types, variadic), false);
```

Comment:
Are you also okay with leaving everything a const std::vector<type> for now and
do all the std::moves() later?  -- I've experimented with some small programs
and clang/gcc produced exactly the same asm for move semantics and const-only
vectors (it may be different of course for the actual code here).